Lib/test/test_shutil.py

```diff
@@ -3365,6 +3365,40 @@ def test_file2file_not_supported(self):
         finally:
             shutil._USE_CP_SENDFILE = True
 
+    def test_exception_on_enodata_call(self):
+        # Test logic when sendfile(2) call returns ENODATA error on
+        # the not-first call on the file and we need to fall back to
+        # traditional POSIX while preserving the position of where we
+        # got to in writing
+        def syscall(*args, **kwargs):
+            if not flag:
+                flag.append(None)
+                return orig_syscall(*args, **kwargs)
+            else:
+                raise OSError(errno.ENODATA, "yo")
```

Comment:
```suggestion
            if flag:
                raise OSError(errno.ENODATA, "yo")
            flag.append(None) 
            return eval(self.PATCHPOINT)(*args, **kwargs)
```

And remove `orig_syscall`.